alert-common/src/main/java/com/synopsys/integration/alert/common/AlertResourceLoader.java

```diff
@@ -0,0 +1,50 @@
+package com.synopsys.integration.alert.common;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AlertResourceLoader {
+    private final AlertProperties alertProperties;
+
+    @Autowired
+    public AlertResourceLoader(final AlertProperties alertProperties) {
+        this.alertProperties = alertProperties;
+    }
+
+    public String getResourceAsString(String path) throws IOException {
+        InputStream inputStream = getInputStreamResource(path);
+        return IOUtils.toString(inputStream, StandardCharsets.UTF_8.name());
+    }
+
+    public InputStream getInputStreamResource(String path) throws IOException {
+        return getResource(path).getInputStream();
+    }
+
+    public Resource getResource(String path) {
+        return new ClassPathResource(path);
+    }
+
+    public String getTemplatePath(String channelDirectory) {
```

Comment:
Do we want the variable to be named `channelDirectory`? I know where this code is coming from, but nothing about the class or method name suggests that it is channel specific.

Reply:
I thought of that as well but couldn't think of a time when we would have a template without a channel associated.